src/framework/ui/menu/menuGroup.component.tsx

```diff
@@ -0,0 +1,244 @@
+const MAIN_ITEM_KEY: string = 'Main Item';
+const SEPARATOR_ELEMENT_KEY: string = 'Separator';
+const SUB_ITEMS_MEASURE_TAG: string = 'Sub Items';
+
+class MenuGroupComponent extends React.Component<MenuGroupProps, ComponentState> {
+
+  static styledComponentName: string = 'MenuGroup';
+
+  public state: ComponentState = {
+    subItemsVisible: false,
+    subItemsHeight: 0,
+  };
+
+  public componentDidUpdate(prevProps: MenuGroupProps, prevState: ComponentState): void {
+    if (prevState.subItemsVisible !== this.state.subItemsVisible) {
+      if (this.state.subItemsVisible) {
+        this.subItemsExpandAnimate(this.state.subItemsHeight);
+        this.animateIcon(0);
+      } else {
+        this.subItemsExpandAnimate(0);
+        this.animateIcon(-180);
+      }
+    }
+  }
+
+  private subItemsAnimation: Animated.Value = new Animated.Value(0);
+  private iconAnimation: Animated.Value = new Animated.Value(-180);
+
+  private subItemsExpandAnimate = (toValue: number): void => {
+    Animated.spring(this.subItemsAnimation, {
+      toValue: toValue,
+    }).start();
+  };
+
+  private animateIcon = (toValue: number): void => {
+    Animated.timing(this.iconAnimation, {
+      toValue: toValue,
+      duration: 200,
+    }).start();
+  };
+
+  private onMainItemPress = (): void => {
+    const subItemsVisible: boolean = !this.state.subItemsVisible;
+
+    this.setState({ subItemsVisible });
+  };
+
+  private onSubItemPress = (index: number, event: GestureResponderEvent): void => {
+    if (this.props.onSelect) {
+      this.props.onSelect(index, event);
+    }
+  };
+
+  private getComponentStyles = (style: StyleType): StyleType => {
+    return {
+      subContainer: {
+        paddingHorizontal: style.subItemsPaddingHorizontal,
+      },
+    };
+  };
+
+  private onSubMenuMeasure = (result: MeasureResult): void => {
+    const subItemsHeight: number = result[SUB_ITEMS_MEASURE_TAG].size.height;
+
+    this.setState({ subItemsHeight });
+  };
+
+  private getIsSelected = (item: MenuItemType): boolean => {
+    const { selectedIndex } = this.props;
+
+    return selectedIndex === item.menuIndex;
+  };
+
+  private renderSeparator = (style: StyleType): React.ReactElement<ViewProps> => {
+    return (
+      <View
+        key={SEPARATOR_ELEMENT_KEY}
+        style={style}
+      />
+    );
+  };
+
+  private renderMainItemAccessory = (style: StyleType): IconElement => {
+    const rotateInterpolate = this.iconAnimation.interpolate({
+      inputRange: [-180, 0],
+      outputRange: ['-180deg', '0deg'],
+    });
+    const animatedStyle: StyleType = { transform: [{ rotate: rotateInterpolate }] };
+
+    return (
+      <Chevron
+        style={style}
+        isAnimated={true}
+        animationStyle={animatedStyle}
+      />
+    );
+  };
+
+  private renderMenuItem = (item: MenuItemType,
+                            isMainItem: boolean,
+                            index: number | string): MenuItemElement => {
```

Comment:
string?

Reply:
It can be used as a value for "key" property for the main group item.